In [2]:
#importing libraries
import pandas as pd
import numpy as np
import os
import matplotlib.pyplot as plt
import seaborn as sns
import scipy

# Work

In [4]:
#creathing path
path = r"C:\Users\logan\Downloads\CareerFoundry\Instacart Basket Analysis"

In [5]:
#importing data
ords_prods_merge = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'orders_products_merged 4.82.pkl'))

In [6]:
customers = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'customers.csv'))

In [7]:
customers

user_id First Name    Surnam  Gender           STATE  Age date_joined  \
0         26711    Deborah  Esquivel  Female        Missouri   48    1/1/2017   
1         33890   Patricia      Hart  Female      New Mexico   36    1/1/2017   
2         65803    Kenneth    Farley    Male           Idaho   35    1/1/2017   
3        125935   Michelle     Hicks  Female            Iowa   40    1/1/2017   
4        130797        Ann   Gilmore  Female        Maryland   26    1/1/2017   
...         ...        ...       ...     ...             ...  ...         ...   
206204   168073       Lisa      Case  Female  North Carolina   44    4/1/2020   
206205    49635     Jeremy   Robbins    Male          Hawaii   62    4/1/2020   
206206   135902      Doris  Richmond  Female        Missouri   66    4/1/2020   
206207    81095       Rose   Rollins  Female      California   27    4/1/2020   
206208    80148    Cynthia     Noble  Female        New York   55    4/1/2020   

        n_dependants fam_status  income  
0                  3    married  165665  
1                  0     single   59285  
2                  2    married   99568  
3                  0     single   42049  
4                  1    married   40374  
...              ...        ...     ...  
206204             1    married  148828  
206205             3    married  168639  
206206             2    married   53374  
206207             1    married   99799  
206208             1    married   57095  

[206209 rows x 10 columns]

In [8]:
#renaming columns
customers.rename(columns={'Surnam': 'Last Name'}, inplace=True)

In [9]:
customers.rename(columns={'STATE': 'State'}, inplace=True)

In [10]:
customers.rename(columns={'n_dependants': 'dependents'}, inplace=True)

In [11]:
customers

user_id First Name Last Name  Gender           State  Age date_joined  \
0         26711    Deborah  Esquivel  Female        Missouri   48    1/1/2017   
1         33890   Patricia      Hart  Female      New Mexico   36    1/1/2017   
2         65803    Kenneth    Farley    Male           Idaho   35    1/1/2017   
3        125935   Michelle     Hicks  Female            Iowa   40    1/1/2017   
4        130797        Ann   Gilmore  Female        Maryland   26    1/1/2017   
...         ...        ...       ...     ...             ...  ...         ...   
206204   168073       Lisa      Case  Female  North Carolina   44    4/1/2020   
206205    49635     Jeremy   Robbins    Male          Hawaii   62    4/1/2020   
206206   135902      Doris  Richmond  Female        Missouri   66    4/1/2020   
206207    81095       Rose   Rollins  Female      California   27    4/1/2020   
206208    80148    Cynthia     Noble  Female        New York   55    4/1/2020   

        dependents fam_status  income  
0                3    married  165665  
1                0     single   59285  
2                2    married   99568  
3                0     single   42049  
4                1    married   40374  
...            ...        ...     ...  
206204           1    married  148828  
206205           3    married  168639  
206206           2    married   53374  
206207           1    married   99799  
206208           1    married   57095  

[206209 rows x 10 columns]

In [12]:
ords_prods_merge['user_id'].nunique()

206209

### Here I can check and see the count for "user_id" (206,209) is the same in both data sets

In [13]:
merged_dataframe = pd.merge(ords_prods_merge, customers, on='user_id', how='inner')

In [14]:
merged_dataframe

product_id                product_name  aisle_id  department_id  \
0                  1  Chocolate Sandwich Cookies        61             19   
1                  1  Chocolate Sandwich Cookies        61             19   
2                907        Premium Sliced Bacon       106             12   
3                907        Premium Sliced Bacon       106             12   
4               1000                    Apricots        18             10   
...              ...                         ...       ...            ...   
32434207       49235         Organic Half & Half        53             16   
32434208       49520      Orange Sparkling Water       115              7   
32434209       49520      Orange Sparkling Water       115              7   
32434210       49520      Orange Sparkling Water       115              7   
32434211       49520      Orange Sparkling Water       115              7   

          prices  order_id  user_id  order_number  orders_day_of_week  \
0            5.8   3139998      138            28                   6   
1            5.8   1977647      138            30                   6   
2           20.0   3160996      138             1                   5   
3           20.0   2254091      138            10                   5   
4           12.9    505689      138             9                   6   
...          ...       ...      ...           ...                 ...   
32434207     1.8   2542346    80734             3                   3   
32434208     7.5   1363278    74506             1                   1   
32434209     7.5   2008002    74506             2                   0   
32434210     7.5   1595973    74506             3                   1   
32434211     7.5   1680478    74506             4                   0   

          order_hour_of_day  ...   order_frequency_flag  First Name  \
0                        11  ...      Frequent customer     Charles   
1                        17  ...      Frequent customer     Charles   
2                        13  ...      Frequent customer     Charles   
3                        14  ...      Frequent customer     Charles   
4                        12  ...      Frequent customer     Charles   
...                     ...  ...                    ...         ...   
32434207                  8  ...  Non-frequent customer       Frank   
32434208                 14  ...       Regular customer         Joe   
32434209                 15  ...       Regular customer         Joe   
32434210                 17  ...       Regular customer         Joe   
32434211                 16  ...       Regular customer         Joe   

          Last Name Gender      State Age  date_joined dependents  fam_status  \
0               Cox   Male  Minnesota  81     8/1/2019          1     married   
1               Cox   Male  Minnesota  81     8/1/2019          1     married   
2               Cox   Male  Minnesota  81     8/1/2019          1     married   
3               Cox   Male  Minnesota  81     8/1/2019          1     married   
4               Cox   Male  Minnesota  81     8/1/2019          1     married   
...             ...    ...        ...  ..          ...        ...         ...   
32434207     Bowers   Male    Alabama  45    9/28/2019          1     married   
32434208     Coffey   Male    Vermont  44    4/16/2018          3     married   
32434209     Coffey   Male    Vermont  44    4/16/2018          3     married   
32434210     Coffey   Male    Vermont  44    4/16/2018          3     married   
32434211     Coffey   Male    Vermont  44    4/16/2018          3     married   

         income  
0         49620  
1         49620  
2         49620  
3         49620  
4         49620  
...         ...  
32434207  41072  
32434208  97209  
32434209  97209  
32434210  97209  
32434211  97209  

[32434212 rows x 31 columns]